src/util/tubemap.js

```diff
@@ -3145,14 +3153,28 @@ function drawNodes(dNodes, groupNode) {
     .on("mouseout", nodeMouseOut)
     .on("dblclick", nodeDoubleClick)
     .on("click", nodeSingleClick)
-    .style("fill", config.transparentNodesFlag ? "none" : "#fff")
-    .style("fill-opacity", config.showExonsFlag ? "0.4" : "0.6")
-    .style("stroke", "black")
+    .style("fill", (d) => colorNodes(d.name)["fill"])
+    .style("fill-opacity", (d) => colorNodes(d.name)["fill-opacity"])
+    .style("stroke", (d) => colorNodes(d.name)["outline"])
```

Comment:
I'm not sure I see a great way around this, but this is going to call the coloring function three times for every node, each time getting every color attribute, and then each time throw away the two irrelevant color attributes and just take the one that is used.

We should think about if there's a way to redesign things to avoid this. Maybe we *should* put the color information actually inside the node objects, because that would simplify this.

Reply:
I think using `styles()` might let us make the call once and apply a whole object.